json-rpc/src/runtime.rs

```diff
@@ -113,30 +122,26 @@ async fn rpc_endpoint(
         .map_err(|_| reject::custom(DatabaseError))?;
 
     let resp = Ok(if let Value::Array(requests) = data {
-        if requests.len() > service.batch_size_limit as usize {
-            let data = ErrorData::ExceedBatchSizeLimit(ExceedBatchSizeLimit {
-                limit: service.batch_size_limit,
-                batch_request_size: requests.len(),
-            });
-            let mut resp = Map::new();
-            resp.insert(
-                "error".to_string(),
-                JsonRpcError::invalid_request_with_data(Some(data)).serialize(),
-            );
+        match service.validate_batch_size_limit(requests.len()) {
+            Ok(_) => {
+                // batch API call
+                let futures = requests.into_iter().map(|req| {
+                    rpc_request_handler(
+                        req,
+                        service.clone(),
+                        Arc::clone(&registry),
+                        ledger_info.clone(),
+                    )
+                });
+                let responses = join_all(futures).await;
```

Comment:
fwiw if any of the futures fail, `join_all` as a whole will fail, so we should be clear on whether what JSON RPC's policy is on batched responses (should it return an error for an entire batch if any of the requests are bad?)